test.js

```diff
@@ -146,6 +146,14 @@ test('Function.prototype.apply with the arguments object', t => {
   t.true(foo('[string|number] <object>', 'bar', {}));
 });
 
+test('allows `this` to be the typeConfig string', t => {
+  function foo () {
+    return argsert.apply('<null|object> [number]', arguments);
```

Comment:
Think so. According to the MDN docs it's been around since ES5 so I'm willing to bet it's been optimized by now. and even if not, I like the semantics: "argsert, apply this check to the arguments"